### PR TITLE
fix(analytics):  throw `Error` if `domElementId` type is not `string`

### PR DESCRIPTION
**Pull request by kvramyasri7**
#### Description of changes

This change removes type check `domElementId` in `autoTackMedia` util function from `record` api

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ export const autoTrackMedia = async (
 		return;
 	}
 
-	if (typeof domElementId === 'string' && !domElementId) {
+	if (!domElementId) {
```
Comment (5d) [inline packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:190]: later at line#197 it cast domElementId type to string. will that work without checking the type infront?

Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:190]: I believe that should be working  as the type of `properties: Record<string, unknown>`. which has  I do not see any red underscore or TS complaints😅.  That is a record which is `{}` can be casted to a string.

Comment (5d) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:190]: ok, just wondering if a user give the value with number `123`. When force type casting to string type, what could happen? I guess browser would just log an error in console. Not sure how react native would handle this.

Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:190]: Very Good Point. umm, RN I did not think about it. 😭 Any tip ?

Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:190]: I believe it does the same what the other logger messages do and just return from there isn't it?

File: packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts [hunk 2]
```diff
@@ -187,13 +188,21 @@ export const autoTrackMedia = async (
...
 		logger.debug(
 			"Missing domElementId field in 'properties' for MediaAutoTrack event type."
 		);
 		return;
 	}
 
+	if (typeof domElementId !== 'string') {
+		throw new AnalyticsError({
```
Comment (5d) [inline packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:199]: For Analytics category, I think we only throw errors for config validation.

**Review by 5d — CHANGES_REQUESTED** [review]

File: packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts (hunk 2 above)
Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:199]: umm, but why is it? any specific reason. Let me go and re-change it from `AnalyticsError` to `AmplifyError`

Comment (5d) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:199]: For Analytics, the APIs are designed in fire and forget style, which is returning void. The customer won't have chance to catch those errors. It's better to just log and return here.

Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts]: Good to know. Thanks Di. Updated.

File: packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts [hunk 3]
```diff
@@ -187,13 +187,21 @@ export const autoTrackMedia = async (
...
-	if (typeof domElementId === 'string' && !domElementId) {
+	if (!domElementId) {
 		logger.debug(
 			"Missing domElementId field in 'properties' for MediaAutoTrack event type."
 		);
 		return;
 	}
 
+	if (typeof domElementId !== 'string') {
+		throw new AmplifyError({
```
Comment (5d) [inline packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:198]: Could you give me an example code how customer could catch the error you thrown here?

Comment (kvramyasri7) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:198]: Web Or RN ?

Comment (5d) [reply packages/analytics/src/providers/personalize/utils/autoTrackMedia.ts:198]: Either of them, I just want to understand how throwing `AmplifyError` would help here for customer to do error handling.

**Pull request closed without merging**